xbmc/addons/Addon.cpp

```diff
@@ -462,6 +462,33 @@ bool CAddon::GetSettingString(const std::string& key,
   return GetSettingValue<CSettingString>(*this, id, key, value);
 }
 
+bool CAddon::GetKodiInstanceSetting(AddonInstanceId id,
+                                    bool& enabled,
+                                    std::string& name,
+                                    bool presenceValidated)
```

Comment:
Is the presenceValidated mandatory? Why not return false if the setting does not exist?
I'd say it makes the api a bit cumbersome for no reason

Reply:
I can remove as value, thought only for cases if in future this becomes called on place where needs the presence of settings.